src/OrleansAWSUtils/Storage/DynamoDBStorage.cs

```diff
@@ -0,0 +1,430 @@
+            {
+                var value = readCapacityUnitsConfig.Split(new[] { '=' }, StringSplitOptions.RemoveEmptyEntries);
+                if (value.Length == 2 && !string.IsNullOrWhiteSpace(value[1]))
+                    readCapacityUnits = int.Parse(value[1]);
+            }
+
+            var writeCapacityUnitsConfig = parameters.Where(p => p.Contains(WriteCapacityUnitsPropertyName)).FirstOrDefault();
+            if (!string.IsNullOrWhiteSpace(writeCapacityUnitsConfig))
+            {
+                var value = writeCapacityUnitsConfig.Split(new[] { '=' }, StringSplitOptions.RemoveEmptyEntries);
+                if (value.Length == 2 && !string.IsNullOrWhiteSpace(value[1]))
+                    writeCapacityUnits = int.Parse(value[1]);
+            }
+        }
+
+        private void CreateClient()
+        {
+            if (service.StartsWith("http://", StringComparison.InvariantCultureIgnoreCase) ||
+                service.StartsWith("https://", StringComparison.InvariantCultureIgnoreCase))
+            {
+                ddbClient = new AmazonDynamoDBClient(new AmazonDynamoDBConfig { ServiceURL = service });
+            }
+            else
+            {
+                var credentials = new BasicAWSCredentials(accessKey, secretKey);
+                ddbClient = new AmazonDynamoDBClient(credentials, new AmazonDynamoDBConfig { ServiceURL = service, RegionEndpoint = AWSUtils.GetRegionEndpoint(service) });
+            }
+        }
+
+        private async Task<TableDescription> GetTableDescription(string tableName)
+        {
+            try
+            {
+                var description = await ddbClient.DescribeTableAsync(tableName);
+                if (description.Table != null)
+                    return description.Table;
+            }
+            catch (ResourceNotFoundException)
+            {
+                return null;
+            }
+            catch (Exception)
+            {
+                throw;
+            }
+            return null;
+        }
+
+        private async Task CreateTable(string tableName, List<KeySchemaElement> keys, List<AttributeDefinition> attributes)
+        {
+            var request = new CreateTableRequest
+            {
+                TableName = tableName,
+                AttributeDefinitions = attributes,
+                KeySchema = keys,
+                ProvisionedThroughput = new ProvisionedThroughput
+                {
+                    ReadCapacityUnits = readCapacityUnits,
+                    WriteCapacityUnits = writeCapacityUnits
+                }
+            };
+
+            try
+            {
+                var response = await ddbClient.CreateTableAsync(request);
+                TableDescription description = null;
+                do
+                {
+                    description = await GetTableDescription(tableName);
+
+                    await Task.Delay(2000);
+
+                } while (description.TableStatus != TableStatus.ACTIVE);
```

Comment:
Nop, it throw an exception and in 99.99% of the cases the loop never execute twice but, AWS docs say that we have to wait for active. I just changed it a bit to make the loop only run while `TableStatus.CREATING`.